### DISCUSSION TITLE
PowerDistribution PanelPoles are made safe in different ways?

### DISCUSSION THREAD

**Discussion by pawsey-kbuckley** [opendcim/openDCIM#1528] [Q&A]
TL';DR Doubt that this is going to affect anyone not using Pawsey-local patches

I think the local issue I saw, is tied into the move to PHP8, however,
in coning to diagnose what was happening, wherin a patch that had
been applied (which does some arithmatic on the chassinum-s and
Panel Poles, to allow for the Panel Pole Scheme in use here to be
displayed) had become broken, I saw that in

```
classes/PowerDistribution.class.php

function MakeSafe(){
    ...
    $this->PanelPole=sanitize($this->PanelPole);
    ...
    $this->PanelPole2=intval($this->PanelPole2);
```
the two PanelPoles appear to be treated as different types,
and so was wondering why ?

**Comment by pawsey-kbuckley** [comment]
Thought to take a look at the database schema, in `create.sql`

```
CREATE TABLE fac_PowerDistribution (
  ...
  PanelPole varchar(20) NOT NULL,
  ...
  PanelPole2 varchar(20) NOT NULL,
  ...
)
```
where they are the same type.

**Comment by wilpig** [comment]
@samilliken and I recently discussed ths and the sanitize function is the correct one.  The reasoning is that outside of the US the poles are often not numeric which is what lead to a PHP8 specific change in 5840213

**Comment by pawsey-kbuckley** [comment]
Funnily enough, after I had fixed up the string<->int issue in our local code,
I then found that I was tripping over the very issue that https://github.com/opendcim/openDCIM/commit/5840213fcbbbc5e897cf4f5614f1e20c965d92b1 addresses, 
however, in the spirit of being explcit as to types, I had added in an extra
explicit type conversion, compared to that commit, at Line 634
in classes/PowerDistribution.class.php, so:

```
$next = strval(intval($this->PanelPole)+$adder);
```
because `$next` is a string, and, even though it may be overkill, using the
explicit `strval()` there, felt like I was leaving less to trip up over.
